lib/process-services/src/lib/task-comments/services/task-comments.service.ts

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AlfrescoApiService, CommentModel, CommentsService, LogService, UserProcessModel } from '@alfresco/adf-core';
+import { ActivitiCommentsApi, CommentRepresentation } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { from, Observable, throwError } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskCommentsService implements CommentsService {
+
+  _commentsApi: ActivitiCommentsApi;
+  get commentsApi(): ActivitiCommentsApi {
+      this._commentsApi = this._commentsApi ?? new ActivitiCommentsApi(this.apiService.getInstance());
+      return this._commentsApi;
+  }
+
+  constructor(
+    private apiService: AlfrescoApiService,
+    private logService: LogService
+  ) {}
+
+  /**
+   * Gets all comments that have been added to a task.
+   *
+   * @param id ID of the target task
+   * @returns Details for each comment
+   */
+  get(id: string): Observable<CommentModel[]> {
+    return from(this.commentsApi.getTaskComments(id))
+    .pipe(
+        map((response) => {
+            const comments: CommentModel[] = [];
+
+            response.data.forEach((comment: CommentRepresentation) => {
+              this.addToComments(comments, comment);
+            });
+
+            return comments;
+        }),
+        catchError(
+          (err: any) => this.handleError(err)
+        )
+    );
+  }
+
+  /**
+   * Adds a comment to a task.
+   *
+   * @param id ID of the target task
+   * @param message Text for the comment
+   * @returns Details about the comment
+   */
+  add(id: string, message: string): Observable<CommentModel> {
+    return from(this.commentsApi.addTaskComment({ message }, id))
+      .pipe(
+          map(
+            (response: CommentRepresentation) => this.newCommentModel(response)
+          ),
+          catchError(
+            (err: any) => this.handleError(err)
+          )
+      );
+  }
+
+  private addToComments(comments: CommentModel[], comment: CommentRepresentation): void {
+    const user = new UserProcessModel(comment.createdBy);
+
+    const newComment: CommentRepresentation = {
+      id: comment.id,
+      message: comment.message,
+      created: comment.created,
+      createdBy: user
+    };
+
+    comments.push(this.newCommentModel(newComment));
+  }
+
+  private newCommentModel(representation: CommentRepresentation): CommentModel {
+    return new CommentModel({
+      id: representation.id,
+      message: representation.message,
+      created: representation.created,
+      createdBy: representation.createdBy
+    });
+  }
+
+  private handleError(error: any) {
+    this.logService.error(error);
```

Comment:
maybe does not make sense, the end application can catch and use log service if needed?